lib/SILOptimizer/Differentiation/JVPCloner.cpp

```diff
@@ -1283,6 +1341,14 @@ class JVPCloner::Implementation final
       retElts.push_back(tanVal);
     }
 
+    // Deallocate local allocations.
+    for (auto alloc : differentialLocalAllocations) {
+      // Assert that local allocations have at least one use.
+      // Buffers should not be allocated needlessly.
+      assert(!alloc->use_empty());
+      diffBuilder.createDeallocStack(diffLoc, alloc);
```

Comment:
I wonder why `destroy_addr` for local allocations isn't needed here, since the corresponding logic in `PullbackCloner` does generate `destroy_addr` for local allocations.

https://github.com/apple/swift/blob/a5ca83a8b2d499ff1dd0bd7693e779a9ce3aca8c/lib/SILOptimizer/Differentiation/PullbackCloner.cpp#L1951-L1961

Do you have any idea why `destroy_addr` isn't needed here? What happens if you add `diffBuilder. emitDestroyAddrAndFold(diffLoc, alloc);`?

Reply:
Not sure, we should probably add more tests with `Tracked<Float>` as in https://github.com/apple/swift/pull/33218

Reply:
Adding tests with `Tracked<Float>` is a great idea!